lib/common/templates/markdown/table.md.template.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @typedef {import('../../../../types/DesignToken.ts').TransformedToken} TransformedToken
+ * @typedef {import('../../../../types/Config.ts').Config} Config
+ * @typedef {import('../../../../types/Config.ts').LocalOptions} LocalOptions
+ */
+
+/**
+ * @param {{
+ *   allTokens: TransformedToken[]
+ *   formatProperty: (token: TransformedToken) => string
+ *   options: Config & LocalOptions
+ *   header: string
+ * }} opts
+ */
+export default ({ allTokens, formatProperty, options, header }) => {
+  const hasDescription = options.showDescriptionColumn;
+
+  return `
+${header}
+
+| Token | ${hasDescription ? 'Description  | ' : ''}Type | Value |
+| --- | ${hasDescription ? '---  | ' : ''}--- | --- |
+${allTokens
+  .map(
+    (token) =>
+      `| ${token.name.replace(/ $/, '')} | ${hasDescription ? (token.$description ? token.$description : token.comment ? token.comment : '') + ' | ' : ''}${token.original.type} | \u0060${options.usesDtcg ? JSON.stringify(token.original.$value) : token.original.value}\u0060 |`,
+  )
+  .join('\n')}
+`;
```

Comment:
Disclaimer: the following code adjustment suggestion was written in Github PR WYSIWYG editor so I can't guarantee that it's valid 😅 

```suggestion
  return `${header.length > 0 ? `${header}\n\n` : ''}| Token | ${hasDescription ? 'Description  | ' : ''}Type | Value |
| --- | ${hasDescription ? '---  | ' : ''}--- | --- |
${allTokens
  .map(
    (token) =>
      `| ${token.name} | ${hasDescription ?
        (
          token.$description ?
            token.$description
            : token.comment ?
              token.comment 
              : ''
        ) + ' | '
      : ''}${options.outputReferences ?  : }${options.usesDtcg ? token.$type : token.type} | \`${JSON.stringify(options.outputReferences ? 
        (
          options.usesDtcg ?
            token.original.$value
            : token.original.value
        )
      : (
          options.usesDtcg ?
            token.$value
            : token.value
        )}\` |`,
  )
  .join('\n')}
`;
```

If we were to allow the columns option and granular control over which columns and what order, this format becomes a little bit more complex, code-wise, my code suggestion here is:
- we should not have leading newline
- we should not have redundant newlines when there is no file header
- we should support both DTCG and non DTCG formatted tokens
- fixed JSON.stringify to apply to both $value and value
- we should respect whatever token name convention the user configured rather than removing the first space (I'm not sure why it does that tbh)
- I think we should take the "type" rather than "original.type" unless we know a specific use case where the type gets changed (e.g. by a preprocessor) and we have a good reason to not use the transformed type over the original type. To be honest, I'm not even sure if the original.type gets set before the preprocessors lifecycle 😅 in that case it shouldn't be possible to have a different value for original.type vs type.
- we only output the original value and not the resolved value if outputReferences is false so I addressed this
- we should format that nested ternary a bit to make it more readable, or abstract it into a little function, on that note I'm curious if Prettier already released --experimental-nested-ternaries https://prettier.io/blog/2023/11/13/curious-ternaries, maybe with this flag it would auto-format the nested ternary. Feel free to add this flag in this PR if it works :).

For the no fileheader (empty string) use-case, I think we should add a test so we can verify by snapshot no redundant newlines are present. A(n integration) test for DTCG formatted tokens also makes sense imo.

When looking at my code suggestion and the fact that we may make it more complicated with the "columns" option and the ordering, I think I would prefer if we split up the formatting per token in functions (formatToken, formatValue, formatDescription), at least for the value and description this seems important for maintainability/readability. I'd probably abstract the table head part into its own function as well while you're at it.